### PR TITLE
Creates public room webhook and adds discord link on room

### PR DESCRIPTION
**Pull request by vdfdev**
Several improvements to try to foster a player community as discussed on #941:
- Link to discord on our room page

image

-  Notifies #lets-play when someone opens a public room

Screenshot from 2021-09-19 00-14-05

- Reduces from 60 mins to 5 mins how long a public room shows up after inactivity, reducing the "ghost room" problem.

#### Checklist

* [x] Use a separate branch in your local repo (not `master`).
* [x] I've read and agree with the [contribution guidelines](https://www.freeboardgames.org/docs/?path=/story/documentation-contributing--page).

### CODE REVIEW

**Comment by ProspectPyxis** [comment]
I like this change! It's a simple way to increase interaction with our Discord server without outright requiring players to log in to play.

A couple quick questions/points on this, though:
- Does the pop-up go away permanently if you click the x, or is it just for the current room and it'll come back the next time? I ask this because I'm not sure if needing to close the pop-up every time you just want to create a private game with friends might get too annoying or not - it might be something to keep an eye on so that it doesn't start feeling *too* intrusive.
- Should the pop-up only show up for the game's host, maybe? Presumably, they'd be the one deciding if they'd like more players to join in with their game, not the players joining in.

These are just suggestions, not must-implements - I'd like to hear what you think.

**Comment by vdfdev** [comment]
For your first point, no, it is not permanent when you click on the x. I hope that it isnt too intrusive if the user doesnt want to close the reminder, but I agree with you ideally it would hide for a few days at least. 

For both of your points, my main concern right now is that there wont be enough players on the #lets-play channel, so I am leaning toward erring on the side of showing it too much than too little (for hosts vs guest, or for hiding forever or not). Also, I think this solution is the simplest for now, let's see how users behave with these changes and then if we get enough players, we can try to make it a bit more intelligent. Or if we don't get enough players at all, we can give up on these reminders.

File: fbg-server/src/rooms/rooms.service.ts [hunk 1]
```diff
@@ -35,37 +37,41 @@ export class RoomsService {
...
     if (!queryRunner) {
       await inTransaction(this.connection, async (queryRunner) => {
-        await this.saveNewRoom(queryRunner, userId, roomEntity);
+        await this.saveNewRoom(queryRunner, user, roomEntity);
       });
       if (room.isPublic) {
         // Cant do this inside a transaction as we need the new room in the notification.
         await this.lobbyService.notifyLobbyUpdate();
+        const doNothing = () => { return; };
+        this.maybeNotifyDiscord(user, roomEntity).then(doNothing, doNothing); // Fire and forget
```
Comment (jasonharrison) [inline fbg-server/src/rooms/rooms.service.ts:49]: I'm just curious, why is the .then(doNothing, doNothing) necessary? What's the difference between this and ignoring the returned promise?

**Review by jasonharrison — APPROVED** [review]

File: fbg-server/src/rooms/rooms.service.ts (hunk 1 above)
Comment (vdfdev) [reply fbg-server/src/rooms/rooms.service.ts]: You were right, this is not necessary.

**Comment by ProspectPyxis** [comment]
Definitely agree with your points. I've sat on this for a bit, and I now definitely see merit in showing the link too much rather than too little. No further complaints from me, in that case!

**Pull request merged**